### ISSUE TITLE
Invalid instance name used when creating IBM VPCs

### ISSUE THREAD

**Issue by LachlanStuart** [lithops-cloud/lithops#537]
I get this error when running with the ibm_vpc backend in "create" mode:

```
2021-01-15 21:37:29,157 - INFO - lithops.invokers[MainThread] - invokers.py:101 - ExecutorID 164c82-24 | JobID M002 - Selected Runtime: metaspace2020/metaspace-lithops:1.8.3 
2021-01-15 21:37:29,177 - INFO - lithops.invokers[MainThread] - invokers.py:106 - Runtime metaspace2020/metaspace-lithops:1.8.3 is not yet installed
2021-01-15 21:37:29,177 - INFO - lithops.standalone.backends.ibm_vpc.ibm_vpc[MainThread] - ibm_vpc.py:53 - IBM VPC client created - Region: eu-de - Host: None
2021-01-15 21:37:29,177 - INFO - lithops.standalone.backends.ibm_vpc.ibm_vpc[MainThread] - ibm_vpc.py:217 - Creating VM instance proxy metaspace2020/metaspace-lithops:1.8.3
2021-01-15 21:37:30,887 - ERROR - root[MainThread] - base_service.py:252 - Invalid json payload provided
Traceback (most recent call last):
  File "/home/lachlan/miniconda3/envs/sm38/lib/python3.8/site-packages/ibm_cloud_sdk_core/base_service.py", line 246, in send
    raise ApiException(
ibm_cloud_sdk_core.api_exception.ApiException: Error: Invalid json payload provided, Code: 400
2021-01-15 21:37:30,888 - WARNING - lithops.standalone.backends.ibm_vpc.ibm_vpc[MainThread] - ibm_vpc.py:119 - Error: Invalid json payload provided, Code: 400
2021-01-15 21:37:30,888 - ERROR - lithops.standalone.backends.ibm_vpc.ibm_vpc[MainThread] - ibm_vpc.py:241 - There was an error trying to create the VM for proxy metaspace2020/metaspace-lithops:1.8.3
Traceback (most recent call last):
  File "/home/lachlan/dev/metaspace/metaspace/engine/sm/engine/annotation_lithops/executor.py", line 201, in map
    futures = executor.map(
  File "/home/lachlan/dev/lithops/lithops/executors.py", line 233, in map
    runtime_meta = self.invoker.select_runtime(job_id, runtime_memory)
  File "/home/lachlan/dev/lithops/lithops/invokers.py", line 107, in select_runtime
    runtime_meta = self.compute_handler.create_runtime(self.runtime_name)
  File "/home/lachlan/dev/lithops/lithops/standalone/standalone.py", line 273, in create_runtime
    backend = self.create(None, 'proxy', runtime)
  File "/home/lachlan/dev/lithops/lithops/standalone/standalone.py", line 334, in create
    backend.create(name_prefix, name_suffix)
  File "/home/lachlan/dev/lithops/lithops/standalone/backends/ibm_vpc/ibm_vpc.py", line 242, in create
    raise e
  File "/home/lachlan/dev/lithops/lithops/standalone/backends/ibm_vpc/ibm_vpc.py", line 236, in create
    instance = self._create_instance(job_key, call_id)
  File "/home/lachlan/dev/lithops/lithops/standalone/backends/ibm_vpc/ibm_vpc.py", line 120, in _create_instance
    raise e
  File "/home/lachlan/dev/lithops/lithops/standalone/backends/ibm_vpc/ibm_vpc.py", line 117, in _create_instance
    response = self.service.create_instance(instance_prototype_model)
  File "/home/lachlan/miniconda3/envs/sm38/lib/python3.8/site-packages/ibm_vpc/vpc_v1.py", line 3401, in create_instance
    response = self.send(request)
  File "/home/lachlan/miniconda3/envs/sm38/lib/python3.8/site-packages/ibm_cloud_sdk_core/base_service.py", line 246, in send
    raise ApiException(
ibm_cloud_sdk_core.api_exception.ApiException: Error: Invalid json payload provided, Code: 400
```

If I add extra logging I can see the API response:
```json
{"errors":[{"code":"bad_field","message":"Invalid json payload provided","target":{"name":"name","type":"field"}}],"trace":"REMOVED"}
```
And request payload:
```json
{
    "keys": [{"id": "REMOVED"}],
    "name": "lithops-proxy-metaspace2020-metaspace-lithops-1.8.3-instance",
    "profile": {"name": "bx2-8x32"},
    "resource_group": {"id": "REMOVED"},
    "vpc": {"id": "REMOVED"},
    "image": {"id": "REMOVED"},
    "zone": {"name": "eu-de-2"},
    "boot_volume_attachment": {
        "delete_volume_on_instance_delete": true,
        "name": "lithops-proxy-metaspace2020-metaspace-lithops-1.8.3-boot",
        "volume": {
            "capacity": 100,
            "iops": 10000,
            "name": "lithops-proxy-metaspace2020-metaspace-lithops-1.8.3-volume",
            "profile": {"name": "10iops-tier"},
        },
    },
    "primary_network_interface": {
        "name": "eth0",
        "subnet": {"id": "REMOVED"},
        "security_groups": [{"id": "REMOVED"}],
    },
}
```
(IDs removed as I don't know what's safe to share)

I suspect the issue is that the instance name contains `.` characters, as it's built from the runtime name which includes a version number. If I try to create an instance through the IBM Cloud UI with that name, it gives me a validation message complaining about the `.`s.

**Comment by gilv** [comment]
@LachlanStuart what is name = "lithops-proxy-metaspace2020-metaspace-lithops-1.8.3-instance"? You don't need this key at all... but I guess you are correct, i think the best to avoid names with "." at all...may be i add this in verify to Lithops to avoid those names. In auto create mode - name is created by Lithops.

**Comment by gilv** [comment]
@LachlanStuart i actually not follow, what is this payload? it was created by you?

**Comment by LachlanStuart** [comment]
@gilv That request payload is generated completely by Lithops and passed to the `ibm_vpc` library. I don't choose the `name`. Lithops builds `name` based on the `runtime` I specify in the config.

The `runtime` includes a version because I want to follow semver for versioning. A lot of Docker images specify versions this way.

**Comment by gilv** [comment]
@LachlanStuart sorry...my fault. We recently changed the naming convention of runtimes.. we now use something like `ibmfunctions/lithops-cf-v38-7:2215dev0` without dots. Dots causes other issues..any chance you create such runtime or i can create for you

**Comment by gilv** [comment]
@LachlanStuart i actually think we have restrictions on runtime names with dots..need to check why it allowed you create one. @JosepSampe am i correct?

**Comment by JosepSampe** [comment]
Restrictions on runtime names with dots are only set for those backends that are built on top of kuberenetes, i.e [Knative](https://github.com/lithops-cloud/lithops/blob/master/lithops/serverless/backends/knative/knative.py#L485) & [Code Engine](https://github.com/lithops-cloud/lithops/blob/master/lithops/serverless/backends/code_engine/code_engine.py#L104). The rest of backends does not have this runtime name restriction

**Comment by gilv** [comment]
@JosepSampe may be it good to have a common for all...and if some has restrictions, then apply restrictions on all?

**Comment by JosepSampe** [comment]
Maybe yes, we only need to modify [this regex](https://github.com/lithops-cloud/lithops/blob/master/lithops/utils.py#L191)

**Comment by LachlanStuart** [comment]
@gilv It would be preferable if the runtime name replaced dots with acceptable characters instead of rejecting them. Dots are used a lot with versioning, and we use the semver versioning scheme wherever possible. A lot of publicly available Docker images also use dots, including `ibmfunctions/action-python-v3.7:1.14.1`. The `:` character is already replaced with `-` for this `name` field, so I think it would make more sense just to change this so that all invalid characters get replaced with `-`.

**Comment by JosepSampe** [comment]
Fixed in #548
